velox/functions/prestosql/ArrayDuplicates.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/container/F14Set.h>
+
+#include "velox/expression/EvalCtx.h"
+#include "velox/expression/Expr.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/lib/LambdaFunctionUtil.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+template <typename T>
+class ArrayDuplicatesFunction : public exec::VectorFunction {
+ public:
+  /// This class implements the array_duplicates query function.
+
+  /// Along with the hash map, we maintain a `hasNull` flag that indicates
+  /// whether null is present in the array.
+  ///
+  /// Zero element copy:
+  ///
+  /// In order to prevent copies of array elements, the function reuses the
+  /// internal elements() vector from the original ArrayVector.
+  ///
+  /// First a new vector is created containing the indices of the elements
+  /// which will be present in the output, and wrapped into a DictionaryVector.
+  /// Next the `lengths` and `offsets` vectors that control where output arrays
+  /// start and end are wrapped into the output ArrayVector.
+
+  ArrayDuplicatesFunction() {}
```

Comment:
this is not needed; the compiler will create a default constructor automatically